modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenClientCredentialsGrantTestCase.java

```diff
@@ -0,0 +1,354 @@
+
+    private static final String INTERNAL_ACTION_MANAGEMENT_VIEW = "internal_action_mgt_view";
+    private static final String INTERNAL_ACTION_MANAGEMENT_CREATE = "internal_action_mgt_create";
+    private static final String INTERNAL_ACTION_MANAGEMENT_UPDATE = "internal_action_mgt_update";
+    private static final String INTERNAL_ACTION_MANAGEMENT_DELETE = "internal_action_mgt_delete";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_LIST = "internal_org_user_mgt_list";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_VIEW = "internal_org_user_mgt_view";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_CREATE = "internal_org_user_mgt_create";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_UPDATE = "internal_org_user_mgt_update";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_DELETE = "internal_org_user_mgt_delete";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_VIEW = "internal_application_mgt_view";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_UPDATE = "internal_application_mgt_update";
+    private static final String INTERNAL_API_RESOURCE_VIEW = "internal_api_resource_view";
+    private static final String INTERNAL_API_RESOURCE_CREATE = "internal_api_resource_create";
+    private static final String CUSTOM_SCOPE_1 = "test_custom_scope_1";
+    private static final String CUSTOM_SCOPE_2 = "test_custom_scope_2";
+    private static final String CUSTOM_SCOPE_3 = "test_custom_scope_3";
+    private static final String NEW_SCOPE_1 = "new_test_custom_scope_1";
+    private static final String NEW_SCOPE_2 = "new_test_custom_scope_2";
+    private static final String NEW_SCOPE_3 = "new_test_custom_scope_3";
+    private static final String NEW_SCOPE_4 = "replaced_scope";
+
+    private static final String SCIM2_USERS_API = "/o/scim2/Users";
+    private static final String ACTIONS_API = "/api/server/v1/actions";
+    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
+    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
+    private static final String MOCK_SERVER_ENDPOINT = "/test/action";
+
+    private SCIM2RestClient scim2RestClient;
+    private String accessToken;
+    private String clientId;
+    private String actionId;
+    private String applicationId;
+    private String domainAPIId;
+    private String userId;
+    private String roleId;
+    private JWTClaimsSet jwtClaims;
+
+    /**
+     * Initializes Test environment and sets up necessary configurations.
+     *
+     * @throws Exception If an error occurs during initialization
+     */
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+        restClient = new ActionsRestClient(serverURL, tenantInfo);
```

Comment:
This will remain unchanged since the super class (ActionsBaseTestCase} instantiates the ActionsRestClient with the variable `restClient`.

Reply:
Can we rename to actionsRestClient. we just added that class.

Also if that class instantiates this why you need to create this again

Reply:
Would that be all right if we address the variable renaming in a follow-up PR? Currently, the ActionsBaseTestCase instantiates the rest client as "restClient". The password grant test case utilizes this when re-instantiating. Since this is already merged shall we make this update immediately after all the gran types are sorted?

We planned to evaluate the necessity of re-instantiating as a follow-up effort as we observed an issue related to the backend URL when the above is not done. We have tracked this for further improvements.